source/crud/tests/v2/bulkWrite-arrayFilters2.yml

```diff
@@ -0,0 +1,37 @@
+runOn:
+    -
+        # arrayFilters support first introduced in 3.5.6
+        maxServerVersion: "3.5.5"
+
+data:
+    - {_id: 1, y: [{b: 3}, {b: 1}]}
+    - {_id: 2, y: [{b: 0}, {b: 1}]}
+
+collection_name: &collection_name "test"
+database_name: &database_name "crud-tests"
+
+tests:
+  -
+    description: "BulkWrite on server (that doesn't support arrayFilters) with arrayFilters on second op"
+    operations:
+      -
+        name: "bulkWrite"
+        arguments:
+          requests:
+            -
+              # UpdateOne with no arrayFilters
+              name: "updateOne"
+              arguments:
+                filter: {}
+                update: { $set: { "y.0.b": 2 } }
+            -
+              # UpdateMany with arrayFilters
+              name: "updateMany"
+              arguments:
+                filter: {}
+                update: { $set: { "y.$[i].b": 2 } }
+                arrayFilters: [ { "i.b": 1 } ]
+          options: { ordered: true }
+        result:
+          # Client-side error.
+          errorContains: The selected server does not support array filters
```

Comment:
This error message is C driver specific. I think we should just assert that an error occurs to keep this driver agnostic. Example:
https://github.com/mongodb/specifications/blob/master/source/crud/tests/v2/bulkWrite-delete-hint-clientError.yml#L61

Reply:
Done